crates/matrix-sdk-crypto/src/olm/group_sessions/inbound.rs

```diff
@@ -55,6 +55,26 @@ use crate::{
     },
 };
 
+/// The trust level required to decrypt an event
+#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+pub enum TrustRequirement {
```

Comment:
I expected 2 options here? 
- Legacy: Like it's doing now
- InvisibleCrypto: The new mode

And a 3rd temporary one for Invisible transition mode?

What do you think, can we find a way to make the encryption mode a bit symetrical to the decryption mode?

Reply:
I just implemented what was given by the "Quarantine" section of https://github.com/matrix-org/matrix-rust-sdk/issues/3544 But I think that "Legacy" would be "Untrusted", and "InvisibleCrypto" would be "CrossSigned" or "CrossSignedOrLegacy". transition would be "CrossSignedOrLegacy".

Reply:
Sorry, I moved that section into a comment here: https://github.com/matrix-org/matrix-rust-sdk/issues/3544#issuecomment-2245005736

Reply:
What is the result of this conversation? Is this resolved or do we need to remove some variants?

Reply:
I've dropped the `VerifiedUserIdentity` option, so we now have three options, which correspond do legacy, invisible crypto, and transition